File: 888.公平的糖果交换.py
#
# @lc app=leetcode.cn id=888 lang=python3
#
# [888] 公平的糖果交换
#

# @lc code=start
class Solution:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
        Sa, Sb = sum(A), sum(B)
        setB = set(B)
        for x in A:
            if x + (Sb - Sa) // 2 in B:
                return (x, x + (Sb - Sa) // 2)
        
# @lc code=end

File: 888.公平的糖果交换.cpp
/*
 * @lc app=leetcode.cn id=888 lang=cpp
 *
 * [888] 公平的糖果交换
 */

// @lc code=start
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& A, vector<int>& B) {
        int suma = 0, sumb = 0;
        for(auto num : A){
            suma += num;
        }
        for(auto num :B){
            sumb += num;
        }
        unordered_map<int, int> target;
        for(auto num : A){
            target[num + (sumb-suma)/2] = num;
        }
        for(auto num :B){
            if(target.find(num) != target.end()){
                return {target[num], num};
            }
        }
        return {};
    }
};
// @lc code=end

